cmd/ipfs/daemon.go

```diff
@@ -459,6 +460,17 @@ func daemonFunc(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment
 
 	printSwarmAddrs(node)
 
+	if node.PrivateKey.Type() == p2pcrypto.RSA {
+		fmt.Print(`
+You are using an RSA Peer ID, thoses are no longer the default recomanded in Kubo since 2020/09.
+They have higher CPU load on both your node and other nodes in the network.
+You should consider changing your public key type to Ed25519 (recommended), ECDSA or secp256k1 key with:
+  ipfs key rotate -o my-old-key -t ed25519
+Then restart your node for it to take effect.
```

Comment:
I don't think this is true, at least not for signature verification. See my latest benchmark in https://github.com/libp2p/go-libp2p/issues/1952#issue-1493205184.

It's still a good idea to use Ed25519, since the keys are way smaller (among other advantages).

Reply:
So, this is true for signing, not for verifying. So this is an excellent reason to rotate keys.

Reply:
ChatGPT  reformatted it to look better in terminal and include info about signing cost :upside_down_face: 

```suggesting
Warning: You are using an RSA Peer ID, which was replaced by Ed25519
as the default recommended in Kubo since September 2020. Signing with
RSA Peer IDs is more CPU-intensive than with other key types.
It is recommended that you change your public key type to ed25519
by using the following command:

  ipfs key rotate -o rsa-key-backup -t ed25519

After changing your key type, restart your node for the changes to
take effect.
```

Reply:
Fixed